File: chor-extraction/src/main/java/extraction/ChoreographyExtraction.java
package extraction;

import antlr4.NetworkLexer;
import antlr4.NetworkParser;
import ast.cc.interfaces.CCNode;
import ast.sp.interfaces.SPNode;
import org.antlr.v4.runtime.ANTLRInputStream;
import org.antlr.v4.runtime.CommonTokenStream;
import org.antlr.v4.runtime.tree.ParseTree;

public class ChoreographyExtraction {

    private ANTLRInputStream stream;
    private NetworkLexer lexer;
    private NetworkParser parser;
    private ParseTree tree;
    private NetworkVisitor networkVisitor;

    public ChoreographyExtraction(String grammar) {
        try {
            stream = new ANTLRInputStream(grammar);
            lexer = new NetworkLexer(stream);
            parser = new NetworkParser(new CommonTokenStream(lexer));
            tree = parser.network();
            networkVisitor = new NetworkVisitor();
        } catch (Exception e) {
            System.out.println(e.getMessage());
        }
    }

    public CCNode extract() throws Exception {
        NetworkExtraction networkExtraction = new NetworkExtraction(networkVisitor.visit(tree));
        return networkExtraction.graphToChoreograpy();
    }

    public SPNode getNetwork(){
        return networkVisitor.visit(tree);
    }
}

File: chor-extraction/src/main/java/extraction/ChoreographyExtraction.kt
package extraction

import ast.cc.interfaces.CCNode
import ast.cc.nodes.Termination
import ast.sp.nodes.Network
import gen.NetworkLexer
import gen.NetworkParser
import org.antlr.v4.runtime.ANTLRInputStream
import org.antlr.v4.runtime.CommonTokenStream

class ChoreographyExtraction {

    @Throws(Exception::class)
    fun extractChoreography(grammar: String): CCNode {
        try {
            val stream = ANTLRInputStream(grammar)
            val lexer = NetworkLexer(stream)
            val parser = NetworkParser(CommonTokenStream(lexer))
            val tree = parser.network()
            val networkVisitor = NetworkVisitor()
            val sp = networkVisitor.visitNetwork(tree) as Network
            val np = NetworkExtraction(sp).extract()
        } catch (e: Exception) {
            println(e.message)
        }

        return Termination()
    }
}
